### PR TITLE
AnimesGG | Add Presence

### PR DESCRIPTION
**Pull request by MMDDKK6500**
**This is an Brazilian website, so the presence is in Brazilian Portuguese.**
This is my second presence for PreMiD, I hope I did everything right.

(The website is kind of slow, so it takes a few seconds for the presence to appear)
Proof that it works:
ezgif com-gif-maker (8)

### CODE REVIEW

File: websites/A/AnimesGG/presence.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    "/calendario/": {
+      details: "Vendo o calendario de animes",
+      state: "Vendo..."
+    }
+  };
+
+presence.on("UpdateData", async () => {
+  let data: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: Math.floor(Date.now() / 1000)
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:30]: Make a timestamp variable and put it outside then use it here

File: websites/A/AnimesGG/presence.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        timestamps = presence.getTimestampsfromMedia(video);
+      data.details = "Assitindo um anime!";
+      data.state = `Vendo ${animeTitleWatching}`;
+      data.smallImageKey = "play";
+      if (video.paused) {
+        delete data.endTimestamp;
+        delete data.startTimestamp;
+        data.smallImageKey = "pause";
+      }
+      [data.startTimestamp, data.endTimestamp] = timestamps;
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:72]:
```suggestion
      [, data.endTimestamp] = timestamps;
```

File: websites/A/AnimesGG/presence.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      video = document.querySelector(
+        "#video > div > div > video"
+      ) as HTMLMediaElement,
+      timestamps = presence.getTimestampsfromMedia(video);
+    data.details = "Assitindo um filme de anime";
+    if (showFullName === true && animeTitleFull !== undefined && animeTitleFull !== null && animeTitleFull !== "") 
+      data.state = `Vendo sobre ${animeTitleFull}`;
+    else 
+      data.state = `Vendo sobre ${animeTitleAbout}`;
+    [data.startTimestamp, data.endTimestamp] = timestamps;
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:116]:
```suggestion
    [, data.endTimestamp] = timestamps;
```

File: websites/A/AnimesGG/presence.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    data.state = "Na lista de animes.";
+    if (params.get("search") !== "0" )
+      data.state = `Procurando por ${params.get("search")}`;
+  } else if (page === "/filmes/") {
+    data.details = "Procurando filmes de animes.";
+    data.state = "Na lista de filmes de animes.";
+    if (params.get("search") !== "0" && params.get("search") !== null)
+      data.state = `Procurando por ${params.get("search")}`;
+  } else if (page.includes("/anime/")) {
+    const isEpisode = page.includes("/episodio");
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:55]: You use it only once, I don't get why you have to make it a variable. To make it more readability? If that's the case then sure I guess

File: websites/A/AnimesGG/presence.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    } else {
+      const animeTitleAbout = document.querySelector(
+          ".ani_titulo > h1"
+        ).innerHTML,
+        animeTitleFull = document.querySelector(
+          ".ani_titulo_original"
+        ).innerHTML,
+        showFullName = await presence.getSetting("fullName");
+      data.details = "Vendo sobre um anime.";
+      if (showFullName === true && animeTitleFull !== undefined && animeTitleFull !== null && animeTitleFull !== "")
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:88]:
```suggestion
      if (showFullName && animeTitleFull) 
```

File: websites/A/AnimesGG/presence.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+      animeTitleFull = document.querySelector(
+        ".ani_titulo_original"
+      ).innerHTML,
+      showFullName = await presence.getSetting("fullName"),
+      video = document.querySelector(
+        "#video > div > div > video"
+      ) as HTMLMediaElement,
+      timestamps = presence.getTimestampsfromMedia(video);
+    data.details = "Assitindo um filme de anime";
+    if (showFullName === true && animeTitleFull !== undefined && animeTitleFull !== null && animeTitleFull !== "")
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:112]:
```suggestion
    if (showFullName && animeTitleFull) 
```

File: websites/A/AnimesGG/presence.ts [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  } else if (page.startsWith("/genero")) {
+    data.details = "Procurando um genero de anime";
+    data.state = `Procurando ${page.substring("/genero/".length).slice(0, -1)}`;
+  } else if (page.startsWith("/lancamentos")) {
+    data.details = "Vendo os ultimos lancamentos no site.";
+    data.state = "Vendo...";
+  }
+
+  if (data.details && data.state)
+    presence.setActivity(data);
```
Comment (Slowlife01) [inline websites/A/AnimesGG/presence.ts:138]:
```suggestion
  if (data.details && data.state) presence.setActivity(data);
  else presence.setActivity();
```

Set the app icon too if you haven't already

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**